data/mods/gen4/moves.ts

```diff
@@ -229,7 +229,9 @@ export const Moves: {[k: string]: ModdedMoveData} = {
 		desc: "The user's type changes based on the battle terrain. Normal type on the regular Wi-Fi terrain. Fails if the user has the Multitype Ability or if the type is one of the user's current types.",
 		shortDesc: "Changes user's type based on terrain. (Normal)",
 		onHit(target) {
-			if (!target.setType('Normal')) return false;
+			// includes() is used here over hasType() because for whatever reason Arceus-Water
```

Comment:
It seems like that should be fixed there probably :S

Reply:
Yeah that'd be fine with me, it's just that I don't know where it's originating the bug from - hasType() calls another function getTypes and it is not obvious at all to me where anything Arceus-related is in here

``getTypes(excludeAdded?: boolean): string[] {
		const types = this.battle.runEvent('Type', this, null, null, this.types);
		if (!excludeAdded && this.addedType) return types.concat(this.addedType);
		if (types.length) return types;
		return [this.battle.gen >= 5 ? 'Normal' : '???'];
	}``

So I had given up originally since I didn't want to start making changes in more broadly called functions without knowing what was going wrong.